contracts/poap/src/contract_tests.rs

```diff
@@ -0,0 +1,962 @@
+            block_info.time = Timestamp::from_seconds(EVENT_START_SECONDS)
+        });
+
+        // Enable mint
+        let msg = ExecuteMsg::EnableMint {};
+        app.execute_contract(
+            Addr::unchecked(ADMIN),
+            poap_contract_addr.clone(),
+            &msg,
+            &vec![],
+        )
+        .unwrap();
+
+        let msg = ExecuteMsg::MintTo {
+            recipient: USER.to_string(),
+        };
+        // Mint the first poap
+        app.execute_contract(
+            Addr::unchecked(ADMIN),
+            poap_contract_addr.clone(),
+            &msg,
+            &vec![],
+        )
+        .unwrap();
+        // Mint the second and last allowed poap
+        app.execute_contract(
+            Addr::unchecked(ADMIN),
+            poap_contract_addr.clone(),
+            &msg,
+            &vec![],
+        )
+        .unwrap();
+
+        let result = app.execute_contract(
+            Addr::unchecked(ADMIN),
+            poap_contract_addr.clone(),
+            &msg,
+            &vec![],
+        );
+        // Should fail since the user have already received the max allowed poaps.
+        assert!(result.is_err());
+
+        // Test also with Mint from use
+        let msg = ExecuteMsg::Mint {};
+        let result = app.execute_contract(
+            Addr::unchecked(USER),
+            poap_contract_addr.clone(),
+            &msg,
+            &vec![],
+        );
+        // Should fail since the user have already received the max allowed poaps.
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn mint_to_only_for_minter_and_admin() {
+        let (mut app, poap_contract_addr) = proper_instantiate();
+        // Change the chain time so that the event is started
+        app.update_block(|block_info| {
+            block_info.time = Timestamp::from_seconds(EVENT_START_SECONDS)
+        });
+
+        let msg = ExecuteMsg::MintTo {
+            recipient: USER.to_string(),
+        };
+        let mint_result = app.execute_contract(
+            Addr::unchecked(USER),
+            poap_contract_addr.clone(),
+            &msg,
+            &vec![],
+        );
+        // User should not be authorized to use the mint to action
+        assert!(mint_result.is_err());
+
+        // Test that minter can call mint to
+        let msg = ExecuteMsg::MintTo {
+            recipient: USER.to_string(),
+        };
+        app.execute_contract(
+            Addr::unchecked(MINTER),
+            poap_contract_addr.clone(),
+            &msg,
+            &vec![],
+        )
+        .unwrap();
+
+        // Test that admin can call mint to
+        let msg = ExecuteMsg::MintTo {
+            recipient: USER.to_string(),
+        };
+        app.execute_contract(
+            Addr::unchecked(ADMIN),
+            poap_contract_addr.clone(),
+            &msg,
+            &vec![],
+        )
+        .unwrap();
+    }
+
+    #[test]
+    fn only_creator_can_change_event_info() {
```

Comment:
It can be split into
1. non_creator_change_event_info_error
2. creator_change_event_info_properly ( need a state check on event info )